.github/workflows/tests.yml

```diff
@@ -0,0 +1,193 @@
+name: tests
+
+on:
+  push:
+  pull_request:
+  schedule:
+    - cron: '0 0 * * *'
+
+jobs:
+  linux_tests:
+
+    runs-on: ubuntu-20.04
+    strategy:
+      matrix:
+        php: ["5.6", "7.0", "7.1", "7.2", "7.3", "7.4"]
+        stability: ["Stable"]
+        include:
+          - php: "5.6"
+            stability: "Lowest"
+          - php: "7.4"
+            stability: "Lowest"
+          - php: "8.0"
+            stability: "Ignore"
+
+    name: PHP ${{ matrix.php }} - ${{ matrix.stability }} - Linux
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ matrix.php }}
+          tools: composer:v2
+          coverage: none
+
+      - name: Setup problem matchers
+        run: echo "::add-matcher::${{ runner.tool_cache }}/phpunit.json"
+
+      - name: Install platform ignoring dependencies
+        uses: nick-invision/retry@v1
+        with:
+          timeout_minutes: 5
+          max_attempts: 5
+          command: composer update --prefer-dist --no-interaction --no-progress --ignore-platform-req=php
+        if: "matrix.stability == 'Ignore'"
+
+      - name: Install stable dependencies
+        uses: nick-invision/retry@v1
+        with:
+          timeout_minutes: 5
+          max_attempts: 5
+          command: composer update --prefer-dist --no-interaction --no-progress
+        if: "matrix.stability == 'Stable'"
+
+      - name: Install lowest dependencies
+        uses: nick-invision/retry@v1
+        with:
+          timeout_minutes: 5
+          max_attempts: 5
+          command: composer update --prefer-stable --prefer-lowest --prefer-dist --no-interaction --no-progress
+        if: "matrix.stability == 'Lowest'"
+
+      - name: Execute tests
+        run: vendor/bin/phpunit
+        if: "matrix.php != '8.0'"
+
+      - name: Execute tests with weak deprecation warnings
+        run: SYMFONY_DEPRECATIONS_HELPER=weak vendor/bin/phpunit
```

Comment:
In my tests I didn't need this `SYMFONY_DEPRECATIONS_HELPER=weak` variable to make PHPUnit pass, are you sure it is required?

Reply:
This is needed for PHP 8.0. The UI looks good with just warnings, but if you check the error code, you will find it is non-zero. ;)

Reply:
Add the env in the matrix for PHP 8.0 then, so 2 steps for tests won't be needed.